hoomd/mpcd/RotatedParallelPlateGeometry.h

```diff
@@ -0,0 +1,176 @@
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar sin_angle = slow::sin(m_angle);
+        const Scalar cos_angle = slow::cos(m_angle);
+        const Scalar tan_angle = sin_angle / cos_angle;
+        const Scalar y_center = pos.x * tan_angle;
+        const Scalar y1 = y_center + m_H;
+        const Scalar y2 = y_center - m_H;
+
+        /* Check if particle is in bounds or could not have left because
+         * normal speed is zero, and exit early. The normal is
+         * n = [-sin(angle), cos(angle)] with no z component.
+         */
+        const signed char sign = (char)((pos.y > y1) - (pos.y < y2));
+
+        const Scalar normal_speed = vel.x * -sin_angle + vel.y * cos_angle;
+        if (sign == 0 || (normal_speed == Scalar(0)))
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        /* Find the time remaining when the particle is collided with wall. This time is computed
+         * using the distance difference between pos.y and y_wall in the normal direction divided by
+         * normal speed. The distance difference = (y_wall - pos.y) * (-cos_angle).
+         */
+        const Scalar y_wall = (sign == 1) ? y1 : y2;
+        dt = (y_wall - pos.y) * (-cos_angle) / normal_speed;
+
+        // backtrack the particle for dt to get to point of contact
+        pos -= vel * dt;
+
+        // update velocity according to bounday conditions
+        // no-slip requires reflection of the tangential components, so vel = -vel + 2V.
+        if (m_no_slip)
+            {
+            const Scalar dV = Scalar(sign * 2) * m_V;
+            vel.x = -vel.x + dV * cos_angle;
+            vel.y = -vel.y + dV * sin_angle;
+            vel.z = -vel.z;
+            }
+        // slip requires reflection of the normal components, so vel = vel - 2n,
+        else
+            {
+            const Scalar dv = Scalar(2) * normal_speed;
+            vel.x -= dv * -sin_angle;
+            vel.y -= dv * cos_angle;
+            }
+
+        return true;
+        }
+
+    //! Check if a particle is out of bounds
+    /*!
+     * \param pos Current particle position
+     * \returns True if particle is out of bounds, and false otherwise
+     */
+    HOSTDEVICE bool isOutside(const Scalar3& pos) const
+        {
+        const Scalar tan_angle = slow::tan(m_angle);
+        const Scalar y_center = pos.x * tan_angle;
+        const Scalar y1 = y_center + m_H;
+        const Scalar y2 = y_center - m_H;
+        return (pos.y > y1 || pos.y < y2);
+        }
+
+    //! Get distance between plates
+    /*!
+     * \returns Distance between plates
+     */
+    HOSTDEVICE Scalar getSeparation() const
+        {
+        return m_H * Scalar(2);
```

Comment:
Equivalent but easier for me to read mathematically
```suggestion
        return Scalar(2) * m_H;
```